pytorch_lightning/callbacks/pruning.py

```diff
@@ -282,33 +290,37 @@ def apply_lottery_ticket_hypothesis(self):
         The ``resample_parameters`` argument can be used to reset the parameters with a new :math:`\theta_z \sim \mathcal{D}_\theta`
         """  # noqa: E501
 
-        def copy_param(new, old, name: str) -> None:
+        def copy_param(new: nn.Module, old: nn.Module, name: str) -> None:
             dst = getattr(new, name)
             src = getattr(old, name)
             if dst is None or src is None or not isinstance(dst, torch.Tensor) or not isinstance(src, torch.Tensor):
-                return
+                return None
             dst.data = src.data.to(dst.device)
 
+        if self._original_layers is None:
+            raise RuntimeError
         for d in self._original_layers.values():
-            copy, names = d["data"], d["names"]
-            if self._resample_parameters and hasattr(copy, "reset_parameters"):
+            copy: Module = d['data']  # type: ignore
+            names: List[Tuple[int, str]] = d["names"]  # type: ignore
```

Comment:
Why ignore these?

Reply:
the dict d is of type `Dict[str, Union[Module, List[Tuple[int,str]]]` here, and mypy complains since theoretically `d['data']` could also be of type `List[Tuple[int, str]]` and `d['names']` could also be of type Module